frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -213,6 +214,15 @@ impl CallDef {
 						},
 					);
 
+				if weight_attrs.is_empty() && dev_mode {
+					// inject a default O(1) weight when dev mode is enabled and no weight has
+					// been specified on the call
+					let empty_weight: syn::Expr =
+						syn::parse(quote::quote!(T::WeightInfo::create()).into())
```

Comment:
```suggestion
						syn::parse(quote::quote!(0).into())
```

Should just be the number 0